samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -194,7 +216,7 @@ public boolean equals(Object obj) {
   @Override
   public String toString() {
```

Comment:
Should `equals` and `hashCode` include the `eventTime` and `arrivalSystemNanoTime`? It seems like if two events with the same content came in at different times, then they might still be considered as different events.

Reply:
different yes, but still equal, right? 
Given that eventTime which pre-existed before adding arrivalTime was not included in either `equals` or `hashcode`, if we include `eventTime` or `arrivalTime` that won't be a backward compatible change

Reply:
I wasn't sure about what semantics we wanted. I think there is a reasonable argument either way. But you make a good point that `eventTime` was already not included, so `arrivalTime` can be not included as well.